packages/taco/src/conditions/base/contract.ts

```diff
@@ -82,7 +107,27 @@ export const contractConditionSchema = rpcConditionSchema
     contractAddress: z.string().regex(ETH_ADDRESS_REGEXP).length(42),
     standardContractType: z.enum(['ERC20', 'ERC721']).optional(),
     method: z.string(),
-    functionAbi: functionAbiSchema.optional(),
+    functionAbi: z
+      .union([
+        functionAbiSchema,
+        z
+          .string()
+          .refine(
+            (abi) => {
+              try {
+                toJsonAbiFormat(abi);
+                return true;
+              } catch (e) {
+                return false;
+              }
+            },
+            {
+              message: 'Invalid Human-Readable ABI format',
+            },
+          )
+          .transform(toJsonAbiFormat),
```

Comment:
Let's move this variable into a separate declaration, just like `functionAbiSchema`